analyzer/src/funTest/kotlin/AnalyzerFunTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2025 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.should
+
+import java.io.File
+import java.io.IOException
+
+import org.ossreviewtoolkit.analyzer.Analyzer.ManagedFileInfo
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+
+class AnalyzerFunTest : WordSpec({
+    "afterResolution()" should {
+        "be called even if resolveDependencies() throws" {
+            val analyzerConfig = AnalyzerConfiguration()
+            val repositoryConfig = RepositoryConfiguration()
+            val analyzer = Analyzer(analyzerConfig)
+            val analysisRoot = File(".").absoluteFile
+
+            val calls = mutableListOf<String>()
+
+            val manager = object : PackageManager("Dummy", "Project", analysisRoot, analyzerConfig, repositoryConfig) {
```

Comment:
I believe defining a `DummyPackageManger` class on the top level of this file would help improving the readability of this test. It could expose a `calls` property, so that the variable in line 42 can be dropped.